### ISSUE TITLE
Surface Book 2: Monitors with Surface Dock

### ISSUE THREAD

**Issue by versipellis** [jakeday/linux-surface#155]
Multi-monitor setup doesn't appear to be working for the SB2 with Surface Dock - two U2518 monitors connected via DP->miniDP->SurfaceDock->SB2. I tried plugging everything in before book as per #23 but that doesn't seem to do anything.

**Comment by jakeday** [comment]
@versipellis What's the output of xrandr for you?

**Comment by versipellis** [comment]
```
Screen 0: minimum 320 x 200, current 3240 x 2160, maximum 8192 x 8192
XWAYLAND0 connected 3240x2160+0+0 (normal left inverted right x axis y axis) 320mm x 210mm
   3240x2160     59.98*+

```

**Comment by versipellis** [comment]
@jakeday I just had a thought - do you know what firmware version your Surface Dock is?

**Comment by npjohnson** [comment]
My Dock is on the newest version, I am on the newest JakeDay kernel, and I run 2 monitors off my dock's Mini-DP outputs, plus 1 off USB.

**Comment by jakeday** [comment]
@versipellis @npjohnson I wonder if this is a wayland vs xorg issue. I run xorg and cannot reproduce.

**Comment by versipellis** [comment]
I believe I tested it on xorg too. I'm out for a week, but I'll double-check again when I'm back.

**Comment by versipellis** [comment]
@jakeday xrandr output on xorg (18.04) - still not working. Getting "No DP signal from device" on both screens.

``Screen 0: minimum 8 x 8, current 3240 x 2160, maximum 32767 x 32767
eDP1 connected primary 3240x2160+0+0 (normal left inverted right x axis y axis) 320mm x 210mm
   3240x2160     59.99*+
DP1 disconnected (normal left inverted right x axis y axis)
DP1-1 disconnected (normal left inverted right x axis y axis)
DP1-2 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
``

**Comment by versipellis** [comment]
Hey @jakeday - any ideas based on my xrandr and dmesg? You mentioned it might've been a configuration issue before?

**Comment by jakeday** [comment]
@versipellis No, everything looks right. I don't see why it wouldn't think it's connected and nothing I do can reproduce it. Very strange.

**Comment by jakeday** [comment]
@versipellis Once you got it all booted up and running and it not working, can you disconnect the screen(s) for at least a minute. Plug it back in, then go to your Screen and Monitor settings and see if it's listed but not enabled? I've had that happen a time or two on Ubuntu 17 and 18.

**Comment by versipellis** [comment]
@jakeday Did that, still not there. Here's a very interesting situation however, I disconnected the dock and used a USB-C to DisplayPort cable instead on a hunch. Screen works.

xrandr below: 

> 
> Screen 0: minimum 8 x 8, current 5800 x 2160, maximum 32767 x 32767
> eDP1 connected primary 3240x2160+0+0 (normal left inverted right x axis y axis) 320mm x 210mm
>    3240x2160     59.99*+
> DP1 connected 2560x1440+3240+0 (normal left inverted right x axis y axis) 550mm x 310mm
>    2560x1440     59.95*+
>    2048x1080     60.00    24.00  
>    1920x1080     60.00    50.00    59.94  
>    1920x1080i    60.00    50.00    59.94  
>    1600x1200     60.00  
>    1280x1024     75.02    60.02  
>    1152x864      75.00  
>    1280x720      60.00    50.00    59.94  
>    1024x768      75.03    60.00  
>    800x600       75.00    60.32  
>    720x576       50.00  
>    720x480       60.00    59.94  
>    640x480       75.00    60.00    59.94  
>    720x400       70.08  
> DP1-1 disconnected (normal left inverted right x axis y axis)
> DP1-2 disconnected (normal left inverted right x axis y axis)
> HDMI1 disconnected (normal left inverted right x axis y axis)
> VIRTUAL1 disconnected (normal left inverted right x axis y axis)

**Comment by versipellis** [comment]
I'm further testing this, and this appears to be an issue with the Surface Dock. Again, I'm on X11 and not Wayland. I connected a monitor over USB-C, and it works perfectly fine as long as the Surface Dock isn't plugged in.

dmesg output:
> [189338.113629] usb 1-1.2: new full-speed USB device number 22 using xhci_hcd
> [189338.248832] usb 1-1.2: not running at top speed; connect to a high speed hub
> [189338.256690] usb 1-1.2: New USB device found, idVendor=0bda, idProduct=5400, bcdDevice= 0.03
> [189338.256692] usb 1-1.2: New USB device strings: Mfr=17, Product=18, SerialNumber=19
> [189338.256694] usb 1-1.2: Product: BillBoard Device
> [189338.256695] usb 1-1.2: Manufacturer: Realtek
> [189338.256696] usb 1-1.2: SerialNumber: 123456789ABCDEFGH

xrandr output:
> Screen 0: minimum 8 x 8, current 3240 x 2160, maximum 32767 x 32767
> eDP1 connected primary 3240x2160+0+0 (normal left inverted right x axis y axis) 320mm x 210mm
>    3240x2160     59.99*+
> DP1 disconnected (normal left inverted right x axis y axis)
> HDMI1 disconnected (normal left inverted right x axis y axis)
> VIRTUAL1 disconnected (normal left inverted right x axis y axis)

**Comment by versipellis** [comment]
@jakeday are you on the Noveau or NVIDIA display drivers?

**Comment by jakeday** [comment]
@versipellis I'm on the i915 only. I don't have e Nvidia installed and I blacklisted noveau.

**Comment by versipellis** [comment]
@jakeday if it may interest you, this ended up being resolved with a very simple fix as in #309.

I think the Surface Dock has issue receiving DP, miniDP, and DVI signals. However, HDMI->miniDP works flawlessly.
